File: 733-flood-fill/flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int[][] directions = new int[][] {{0, 1}, {1, 0}, {-1, 0}, {0, -1}};
        int sourceColor = image[sr][sc];
        if (sourceColor == color) return image;
        Queue<int[]> queue = new ArrayDeque<>();
        queue.add(new int[]{sr, sc});
        while (!queue.isEmpty()) {
            int[] curr = queue.poll();
            image[curr[0]][curr[1]] = color;
            for (int[] dir : directions) {
                int nexti = curr[0] + dir[0];
                int nextj = curr[1] + dir[1];
                if (nexti >= 0 && nexti < image.length && nextj >= 0 && nextj < image[0].length && image[nexti][nextj] == sourceColor) {
                    queue.add(new int[]{nexti, nextj});
                }
            }
        }
        return image;
    }
}

File: 733-flood-fill/flood-fill.py
class Solution(object):
    def floodFill(self, image, sr, sc, color):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type color: int
        :rtype: List[List[int]]
        """
        original = image[sr][sc]
        if color == original: return image
        def fill(sr, sc):
            if image[sr][sc] == original:
                image[sr][sc] = color
                if sr - 1 >= 0:
                    fill(sr - 1, sc)
                if sr + 1 < len(image):
                    fill(sr + 1, sc)
                if sc - 1 >= 0:
                    fill(sr, sc - 1)
                if sc + 1 < len(image[0]):
                    fill(sr, sc + 1)
        fill(sr, sc)
        return image
        
        